drivers/net/wireless/mediatek/mt76/mt7921/mac.c

```diff
@@ -1507,3 +1509,45 @@ int mt7921_mac_set_beacon_filter(struct mt7921_phy *phy,
 
 	return 0;
 }
+
+void mt7921_coredump_work(struct work_struct *work)
+{
+	struct mt7921_dev *dev;
+#define SZ_CD (128 * 1024)
+	char *dump = vzalloc(SZ_CD), *data;
+
+	dev = (struct mt7921_dev *)container_of(work, struct mt7921_dev,
+						cd.work.work);
+
+	if (time_is_after_jiffies(dev->cd.last_activity + msecs_to_jiffies(200)))
+		goto out;
+
```

Comment:
I think you can move the buffer allocation after this check. Moreover you do not need the goto since it is only used here

Reply:
ack